torch/csrc/autograd/autograd_meta.cpp

```diff
@@ -175,9 +166,9 @@ void AutogradMeta::set_fw_grad(const at::TensorBase& new_grad_base, const at::Te
 
     // Enforce the basic layout constraint
     if (!has_same_meta(new_grad, self)) {
-      Tensor new_grad_with_meta = new_with_same_meta(self);
-      new_grad_with_meta.copy_(new_grad);
-      new_grad = new_grad_with_meta;
+      auto res = at::_new_zeros_with_same_meta(new_grad, self);
```

Comment:
I'm confused: let's say that new_grad is a BatchedTensor that wraps a tensor of size [B, 2, 2] (call this new_grad_physical) and self is a tensor of size [2, 2]. What does it mean for the new_grad to have the same metadata as self? Does it mean that each slice of new_grad_physical (e.g. new_grad_physical[0]) has the same strides as `self`?

Reply:
I think we discussed this offline, resolving this for now.